hopsworks-api/src/main/java/io/hops/hopsworks/api/admin/SystemAdminService.java

```diff
@@ -91,4 +99,54 @@ public Response changeMasterEncryptionPassword(@Context SecurityContext sc, @Con
           "password file: " + ex.getMessage());
     }
   }
+  
+  @PUT
+  @Path("/variables/refresh")
+  public Response refreshVariables(@Context SecurityContext sc, @Context HttpServletRequest request)
+    throws AppException {
+    LOG.log(Level.FINE, "Requested refreshing variables");
+    settings.refreshCache();
+    
+    JsonResponse response = buildSuccessfulResponse(Response.Status.NO_CONTENT, "Variables refreshed");
+    return noCacheResponse.getNoCacheResponseBuilder(Response.Status.OK).entity(response).build();
+  }
+  
+  @PUT
```

Comment:
Since you are not supplying the complete set of variables (as far as I understand), this is more semantically close to a POST. A PUT should contain all the variables and effectively replace the server side state completely.